array.c

```diff
@@ -6863,23 +6873,28 @@ rb_ary_compact(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.count                   -> int
- *     ary.count(obj)              -> int
- *     ary.count {|item| block}    -> int
+ *    array.count -> an_integer
+ *    array.count(obj) -> an_integer
+ *    array.count {|element| ... } -> an_integer
  *
- *  Returns the number of elements.
+ *  Returns a count of specified elements.
  *
- *  If an argument is given, counts the number of elements which equal +obj+
- *  using <code>==</code>.
+ *  With no argument and no block, returns the count of all elements:
+ *    [0, 1, 2].count # => 3
+ *    [].count # => 0
  *
- *  If a block is given, counts the number of elements for which the block
- *  returns a true value.
+ *  With argument +obj+, returns the count of elements <tt>eql?</tt> to +obj+:
+ *    [0, 1, 2, 0].count(0) # => 2
+ *    [0, 1, 2].count(3) # => 0
  *
- *     ary = [1, 2, 4, 2]
- *     ary.count                  #=> 4
- *     ary.count(2)               #=> 2
- *     ary.count {|x| x%2 == 0}   #=> 3
+ *  With no argument and a block given, calls the block with each element;
+ *  returns the count of elements for which the block returns a truthy value:
+ *    [0, 1, 2, 3].count {|element| element > 1} # => 2
  *
+ *  With argument +obj+ and a block given, issues a warning
+ *  (warning: given block not used); ignores the block;
```

Comment:
We can probably just say it issues a warning and ignores the block, we don't need to provide the warning message nor an example of this case.